### DISCUSSION TITLE
matchPackageNames glob not working as expected

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#30411] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I had this rule:
```json
		{
			"description": "Disable updating target Kotlin version to stay specific to old Gradle.",
			"matchPackagePatterns": [
				"^org\\.jetbrains\\.kotlin:kotlin-.*$"
			],
			"matchFileNames": [
				"gradle/libs.versions.toml"
			],
			"matchCurrentVersion": "1.4.32",
			"matchUpdateTypes": ["major", "minor"],
			"enabled": false
		},
```
which was auto-migrated (by running v38 validator in strict mode which outputs oldconfig/newconfig) as
```patch
-"matchPackagePatterns": [ "^org\\.jetbrains\\.kotlin:kotlin-.*$" ]
+"matchPackageNames": [ "/^org\\.jetbrains\\.kotlin:kotlin-.*$/" ]
```
which I then simplified as
```json
"matchPackageNames": [ "org.jetbrains.kotlin:kotlin-*" ]
```
in this PR: https://github.com/TWiStErRob/net.twisterrob.gradle/pull/780/files
based on docs: https://docs.renovatebot.com/configuration-options/#matchpackagenames

After merging that PR, the package rule broke and Renovate opened and upgrade PR:
https://github.com/TWiStErRob/net.twisterrob.gradle/pull/781

What did I do wrong with globbing? It seems such a simple use case.

Related: https://github.com/isaacs/minimatch/issues/237

### Logs (if relevant)

[scratch_58.txt](https://github.com/user-attachments/files/16398046/scratch_58.txt)

**Comment by rarkins** [comment, marked as answer]
@TWiStErRob the hosted app isn't running v38 yet. So with v37 that will be treated as an exact match string.

We plan to do a staged rollout starting tomorrow, would you like your account to be included?

**Reply by TWiStErRob** [reply]
Aaah, makes sense.
Sounds good to include @TWiStErRob org.

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/discussions/30417
